controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -89,6 +93,8 @@ type Workload struct {
 	etcdClientGenerator etcdClientFor
 }
 
+var _ WorkloadCluster = (*Workload)(nil)
```

Comment:
nit: we are usually creating an empty strct instead of casting nil
```suggestion
var _ WorkloadCluster = &Workload{}
```

Reply:
Took this idea from [capz](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f591d6077623e77388901465e0b7aa8bb40cbeea/cloud/services/groups/client.go#L41) to avoid memory allocation for the struct.

Reply:
+1 for being consistent and use empty struct